nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -144,6 +144,9 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
 
         self.enc_dec_model.model_type = ModelType.encoder_and_decoder
 
+        # Override limit_val_batches to be a multiple of num microbatches to prevent val_step from exiting in between a step
+        self._reconfigure_val_batches()
```

Comment:
are we sure no other child class ends up calling this init?

Reply:
there are too many child classes for LM enc dec, so need to ensure we keep an explicit track...

Reply:
Yes I have checked for that. Wrt `MegatronLMEncoderDecoderModel` it's inherited by `MegatronT5Model` and `MegatronNMTModel`. `MegatronT5Model` calls `super().__init__()` so I haven't added reconfiguring in the T5 model since it's taken care in the parent class.  `MegatronNMTModel` uses batch in its val_step and also has it's own __init__ where we are not doing any reconfiguring.

Reply:
perfect! just wanted to be sure

Reply:
oh btw, what about `MegatronT5FinetuneModel` and its derivatives?
and `MegatronT5PromptLearningModel`?

Reply:
Yup they are taken care as well.

Reply:
wouldn't the `super.__init__` call in `MegatronT5FinetuneModel` mess things? `MegatronT5FinetuneModel` samples global batches so we _dont_ want it to call `_reconfigure_val_batches`, but it looks like it will

Reply:
Yes you are right, this won't break but will end up running num_microbatches times more validation_step than what it is.

Reply:
@aklife97 bcoz of this issue so that child classes that don't need reconfiguring don't end up calling it via `__init__` (I had a `BioMegatron BERT` CI test fail bcoz of this. It din't have global_batch_size and couldn't calculate `get_num_microbatches`), I am now calling reconfiguring in `build_train_valid_test_datasets` method of the models that need reconfiguring (basically pretraining models: `MegatronGPTModel`, `MegatronBERTModel`, `MegatronT5Model`). All other models that use `dataloader_iter` are either passing one batch or they have  build_train_valid_test_datasets implemented in the child class where reconfiguring is handled ex: `MegatronLMEncoderDecoderModel` that doesn't have `build_train_valid_test_datasets`. Or another case is, the model needs reconfiguring (ex: `MegatronBARTModel`) but it uses build_train_valid_test_datasets of the parent class where reconfiguring is done. The CI is now passing fully. Let me know if this is okay.